core/src/main/kotlin/transformers/documentation/DefaultDocumentableMerger.kt

```diff
@@ -2,18 +2,27 @@
  * Copyright 2014-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 
-package org.jetbrains.dokka.base.transformers.documentables
+package org.jetbrains.dokka.transformers.documentation
 
 import org.jetbrains.dokka.DokkaConfiguration
-import org.jetbrains.dokka.base.utils.firstNotNullOfOrNull
+import org.jetbrains.dokka.InternalDokkaApi
 import org.jetbrains.dokka.model.*
 import org.jetbrains.dokka.model.properties.ExtraProperty
 import org.jetbrains.dokka.model.properties.MergeStrategy
 import org.jetbrains.dokka.model.properties.mergeExtras
 import org.jetbrains.dokka.plugability.DokkaContext
-import org.jetbrains.dokka.transformers.documentation.DocumentableMerger
+import org.jetbrains.dokka.CoreExtensions
 
-internal class DefaultDocumentableMerger(val context: DokkaContext) : DocumentableMerger {
+/**
+ * Should NOT be used outside of Dokka itself, there are no guarantees
+ * this class will continue to exist in future releases.
+ *
+ * This class resides in core because it is a non-trivial implementation
+ * for a core extension [CoreExtensions.documentableMerger], which is needed
+ * in modules that only have access to `dokka-core`.
+ */
+@InternalDokkaApi
+public class DefaultDocumentableMerger(context: DokkaContext) : DocumentableMerger {
```

Comment:
I moved this class to `core` since I need it for bootstrapping analysis tests.

Don't really understand why we need the `DocumentableMerger` extension point at all - I find it very difficult to believe someone will need to re-implement it or will even understand how it works (I don't). Should return to it some time in the future when we revisit all core extensions.